plugins/MultiSites/Dashboard.php

```diff
@@ -257,41 +263,39 @@ private function makeCloneOfDataTableSites(DataTable $sites)
      *    site7
      * )
      *
-     * @param $sites
-     * @return array
+     * in a sorted order
+     *
+     * @param DataTable $table
+     * @param array $request
      */
-    private function makeSitesFlat($sites)
+    private function makeSitesFlatAndApplyGenericFilters(DataTable $table, $request)
     {
-        $flatSites = array();
+        // we handle limit here as we have to apply sort filter, then make sites flat, then apply limit filter.
+        $filterOffset = $request['filter_offset'];
+        $filterLimit  = $request['filter_limit'];
+        unset($request['filter_offset']);
+        unset($request['filter_limit']);
 
-        foreach ($sites as $site) {
-            if (!empty($site['subtable'])) {
-                if (isset($site['idsubdatatable'])) {
-                    unset($site['idsubdatatable']);
-                }
-
-                $subtable = $site['subtable'];
-                unset($site['subtable']);
-                $flatSites[] = $site;
-                foreach ($subtable as $siteWithinGroup) {
-                    $flatSites[] = $siteWithinGroup;
-                }
-            } else {
-                $flatSites[] = $site;
-            }
+        // filter_sort_column does not work correctly is a bug in MultiSites.getAll
+        if (!empty($request['filter_sort_column']) && $request['filter_sort_column'] === 'nb_pageviews') {
+            $request['filter_sort_column'] = 'Actions_nb_pageviews';
+        } elseif (!empty($request['filter_sort_column']) && $request['filter_sort_column'] === 'revenue') {
+            $request['filter_sort_column'] = 'Goal_revenue';
         }
 
-        return $flatSites;
-    }
+        // make sure no limit filter is applied, we will do this manually
+        $table->disableFilter('Limit');
 
-    private function applyLimitIfNeeded($sites, $limit)
```

Comment:
This code is no longer needed as well since the `nestedLimit` filter does now apply the limit filter